### PR TITLE
Fix #1247: URL parameters are not escaped.

### LINKED ISSUE #1247: URL parameters are not escaped in many methods
In some cases this makes it impossible to pass certain arguments to APIs. In other cases, if a user of this client is handling untrusted user input, it can be a security issue since end users would be able to inject arbitrary unintended parameters into URLs.

If you look at the exec function in https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java the code for putting command arguments in the URL is currently written as:
```java
    public ExecWatch exec(String... command) {
        StringBuilder sb = new StringBuilder();
        String[] actualCommands = command.length >= 1 ? command : EMPTY_COMMAND;

        sb.append("exec?command=");

        boolean first = true;
        for (String cmd : actualCommands) {
            if (first) {
                first = false;
            } else {
                sb.append("&command=");
            }
            sb.append(cmd);
        }
```

So if you call it with:
```java
exec("/bin/sh", "-c", "echo test 2>&1");
```
the command errors out because `&` gets interpreted by the api servers as starting another query parameter and `2>` alone is invalid shell syntax.

Exec isn't the only place this is an issue, it happens in a lot of other places too. Is there a reason to construct  URLs manually with a StringBuilder instead of using okhttp's HttpUrl.Builder class? http://square.github.io/okhttp/3.x/okhttp/okhttp3/HttpUrl.Builder.html

### PR DESCRIPTION
**Pull request by dev-gaur**
#1247

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java [hunk 1]
```diff
@@ -169,6 +169,39 @@ public void onClose(int i, String s) {
...
+          logger.info("Shell barfed");
+          execLatch.countDown();
+        }
+
+        @Override
+        public void onClose(int i, String s) {
+          logger.info("Shell closed");
+          execLatch.countDown();
+        }
+      }).exec("echo 'H$ll* (W&RLD}'"); // These arguments in the exec command should be escaped.
```
Comment (manusa) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java]: I think you need to separate the arguments. 
Also the `}` in `(W&RLD}` looks like a typo

Comment (dev-gaur) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java]: No I added that bracket just to demonstrated that all of these characters are being escaped properly.

Comment (manusa) [reply kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java]: I understand, but then the assertion was inconsistent (changed now).

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**